packages/kbn-unified-data-table/src/components/data_table_columns.tsx

```diff
@@ -133,13 +135,16 @@ function buildEuiGridColumn({
     editField &&
     dataViewField &&
     buildEditFieldButton({ hasEditDataViewPermission, dataView, field: dataViewField, editField });
-  const columnDisplayName =
+
+  const columnNameWithoutLabel =
     columnName === '_source'
       ? i18n.translate('unifiedDataTable.grid.documentHeader', {
           defaultMessage: 'Document',
         })
       : dataViewField?.displayName || columnName;
 
+  const columnDisplayName = columnHeaders ? columnHeaders[columnName] : columnNameWithoutLabel;
```

Comment:
I believe we can move that logic to a function now:

```js
const getColumnDisplayName = (columnName, dataViewField, columnHeaders) => {
  if (columnHeaders?.[columnName]) {
    return columnHeaders[columnName];
  }

  if (columnName === '_source') {
    return i18n.translate('unifiedDataTable.grid.documentHeader', {
      defaultMessage: 'Document',
    });
  }

  return dataViewField?.displayName || columnName;
};
```

This way we declare more the intent of `columnHeaders` taking precedence of dataViewField.displayName.

and then you can use it the following way:

```js
const columnDisplayName = getColumnDisplayName(columnName, dataViewField, columnHeaders);
```